src/components/PokemonContainer/PokemonContainer.tsx

```diff
@@ -1,151 +1,133 @@
-import { Component } from 'react';
-import type { ChangeEvent } from 'react';
+import { useState, useEffect } from 'react';
+import { useSearchParams } from 'react-router';
 import { SearchBar } from '../SearchBar/SearchBar';
 import PokemonResults from '../PokemonSearchBarResults/PokemonResults';
-import type { Pokemon, PokemonShort } from '../../types_interfaces/interfaces';
+import type { Pokemon } from '../../types_interfaces/interfaces';
 import ErrorBoundaryButton from '../ErrorBoundary/ErrorBoundaryButton';
 import './PokemonContainer.css';
 import Pagination from '../Pagination/Pagination';
+import { fetchPokemonByName, fetchPokemonsPage } from '../../api/pokemon';
+import { useSearchTerm } from '../../Hooks/useSerchTerm';
+import PokemonDetails from '../PokemonDetails/PokemonDetails';
 
-type State = {
-  term: string;
-  loading: boolean;
-  error: string | null;
-  currentPokemon: Pokemon | null;
-  allPokemons: Pokemon[];
-  currentPage: number;
-};
-type Props = {
-  title?: string;
-  value?: string;
-  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
-  func?: (inputValue: string) => void;
-};
+const PokemonContainer: React.FC = () => {
+  const [term, setTerm] = useSearchTerm();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [currentPokemon, setCurrentPokemon] = useState<Pokemon | null>(null);
+  const [allPokemons, setAllPokemons] = useState<Pokemon[]>([]);
+  const [searchParams, setSearchParams] = useSearchParams(); // for URL
 
-export class PokemonContainer extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      term: '',
-      loading: false,
-      error: null,
-      currentPokemon: null,
-      allPokemons: [],
-      currentPage: 1,
-    };
-  }
+  const currentPage = Number(searchParams.get('page')) || 1;
 
-  componentDidMount() {
-    const savedTerm = localStorage.getItem('searchTerm') || '';
-    this.setState({ term: savedTerm }, () => {
-      if (savedTerm.trim()) {
-        this.fetchPokemon(savedTerm.trim());
-      } else {
-        this.fetchAllPokemons();
-      }
-    });
-  }
+  const selectedName = searchParams.get('selected');
 
-  handleChange = (e: ChangeEvent<HTMLInputElement>) => {
-    this.setState({ term: e.target.value });
-  };
+  useEffect(() => {
+    if (term.trim()) {
+      fetchPokemon(term.trim());
+    } else {
+      fetchAllPokemons(currentPage);
+    }
+  }, [currentPage]);
 
-  handleSearch = () => {
-    const trimmed = this.state.term.trim().toLowerCase();
-    localStorage.setItem('searchTerm', trimmed);
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTerm(e.target.value);
+  };
 
+  const handleSearch = () => {
+    const trimmed = term.trim().toLowerCase();
+    setSearchParams({}); // refresh
```

Comment:
search params should be in URL and after that I can reload page and see search result